.github/workflows/upgrade-test.yml

```diff
@@ -0,0 +1,144 @@
+
+          kubectl create -f ${{ env.TEMP_DIR }}/catalogsource.yaml
+
+          echo Wait for CatalogSource ${{ env.CATALOG_SOURCE_NAME }} to start
+          kubectl wait -n ${{ env.CATALOG_SOURCE_NAMESPACE }} catalogsource/${{ env.CATALOG_SOURCE_NAME }} --for=jsonpath='{.status.connectionState.lastObservedState}'=READY --timeout=300s
+
+          kubectl create -f ${{ env.TEMP_DIR }}/subscription.yaml
+
+          echo Waiting for Subscription to be ready
+          kubectl wait -n ${{ env.SUBSCRIPTION_NAMESPACE }} subscription/${{ env.SUBSCRIPTION_NAME }} --for=jsonpath='{.status.state}'=AtLatestKnown --timeout=180s
+
+          echo Waiting for Deployment to be ready
+          timeout 60 bash -c 'until [[ $(kubectl get deployment/opendatahub-operator-controller-manager -n '${{ env.SUBSCRIPTION_NAMESPACE }}') ]]; do sleep 5 && echo "$(kubectl get deployment/opendatahub-operator-controller-manager -n '${{ env.SUBSCRIPTION_NAMESPACE }}')"; done'
+          kubectl wait -n ${{ env.SUBSCRIPTION_NAMESPACE }} deployment/opendatahub-operator-controller-manager --for=condition=Available=true --timeout=60s
+        env:
+          CATALOG_SOURCE_NAME: "odh-olm-test"
+          CATALOG_SOURCE_NAMESPACE: "olm"
+          SUBSCRIPTION_NAME: "opendatahub-operator"
+          SUBSCRIPTION_NAMESPACE: "openshift-operators"
+
+      - name: Create Namespace for DSPO
+        run: |
+          kubectl create namespace ${{ env.DSPO_NS }}
+          kubectl config set-context --current --namespace=${{ env.DSPO_NS }}
+        env:
+          DSPO_NS: data-science-pipelines-operator
+
+      - name: Prepare for Upgrade Testing
+        run: |
+          # Update the KfDef manifest with the latest released version
+          sed -i "s/main/${{ inputs.released-version }}/" ${{ env.RESOURCES_DIR }}/kfdef/kfdef.yaml
+        working-directory: ${{ env.RESOURCES_DIR }}/kfdef
+
+      - name: Print KfDef Manifest Contents
+        run: cat ${{ env.RESOURCES_DIR }}/kfdef/kfdef.yaml
+
+      - name: Deploy DSPO KfDef
+        run: |
+          kubectl apply -f ${{ env.RESOURCES_DIR }}/kfdef/kfdef.yaml -n ${{ env.DSPO_NS }}
+        env:
+          DSPO_NS: "data-science-pipelines-operator"
+
+      - name: Print ODH Operator Pod Logs
+        run: kubectl get pods -n openshift-operators -o jsonpath='{.items[*].metadata.name}' | xargs -I {} kubectl logs -n openshift-operators {}
+
+      - name: Wait for DSPO deployment to be ready
+        run: |
+          kubectl get deployments -n ${{ env.DSPO_NS }}
+          timeout 300 bash -c 'until [[ $(kubectl get deployment/data-science-pipelines-operator-controller-manager -n '${{ env.DSPO_NS }}') ]]; do sleep 5 && echo "$(kubectl get deployment/data-science-pipelines-operator-controller-manager -n '${{ env.DSPO_NS }}')"; done'
+        env:
+          DSPO_NS: data-science-pipelines-operator
+
+      - name: Run pre-upgrade tests
+        run: |
+          chmod +x tests/upgrades/main.sh
```

Comment:
you don't need to run this here, just commit the file with these file permissions

Reply:
Done, updated the commit with these file permissions and removed it from here.